File: src/main/java/org/intellij/sdk/language/jass/JassCommenter.java
package org.intellij.sdk.language.jass;

import com.intellij.lang.Commenter;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

final class JassCommenter implements Commenter {

    @Override
    public @NotNull String getLineCommentPrefix() {
        return "#";
    }

    @Override
    public @NotNull String getBlockCommentPrefix() {
        return "";
    }

    @Nullable
    @Override
    public String getBlockCommentSuffix() {
        return null;
    }

    @Nullable
    @Override
    public String getCommentedBlockCommentPrefix() {
        return null;
    }

    @Nullable
    @Override
    public String getCommentedBlockCommentSuffix() {
        return null;
    }

}

File: src/main/kotlin/raft/war/language/jass/JassCommenter.kt
package raft.war.language.jass

import com.intellij.lang.Commenter

class JassCommenter : Commenter {
    override fun getLineCommentPrefix(): String = "//"

    override fun getBlockCommentPrefix(): String? = null

    override fun getBlockCommentSuffix(): String? = null

    override fun getCommentedBlockCommentPrefix(): String? = null

    override fun getCommentedBlockCommentSuffix(): String? = null
}
